docs/installation.rst

```diff
@@ -289,6 +299,100 @@ Here's the list of the subpackages and what they enable:
+different providers. Again, the general approach her is that the providers are backwards compatible,
+including cross-dependencies. Any kind of breaking changes and requirements on particular versions of other
+provider packages are automatically documented in the release notes of every provider.
+
+.. note::
+    We also provide ``apache-airflow-backport-providers`` packages that can be installed for Airflow 1.10.
+    Those are the same providers as for 2.0 but automatically back-ported to work for Airflow 1.10. Those
+    backport providers are going to be updated and released for 3 months after Apache Airflow 2.0 release.
+
+Provider packages functionality
+-------------------------------
+
+Separate provider packages provide the possibilities that were not available in 1.10:
+
+1. You can upgrade to latest version of particular providers without the need of Apache Airflow core upgrade
+
+2. You can downgrade to previous version of particular provider in case the new version introduces
+   some problems, without impacting the main Apache Airflow core package.
+
+3. You can release and upgrade/downgrade provider packages incrementally, independent from each other. This
+   means that you can incrementally validate each of the provider package update in your environment,
+   following the usual tests you have in your environment.
+
+
+Q&A for provider packages
+-------------------------
+
+Q. When upgrading to a new Airflow version such as 2.0, but possibly 2.0.1 and beyond, is the best practice
+   to also upgrade provider packages at the same time?
+
+A. It depends on your use case. If you have automated or semi-automated verification of your installation,
+   that you can run a new version of Airflow including all provider packages, then definitely go for it.
+   If you rely more on manual testing, it is advised that you upgrade in stages. Depending on your choice
+   you can either upgrade all used provider packages first, and then upgrade Airflow Core or the other way
+   round. The first approach - when you first upgrade all providers is probably safer, as you can do it
+   incrementally, step-by-step replacing provider by provider in your environment.
+
+
+Q. I have an Airflow version (1.10.12) running and it is stable. However, because of a Cloud provider change,
+   I would like to upgrade the provider package. If I don't need to upgrade the Airflow version anymore,
+   how do I know that this provider version is compatible with my Airflow version?
+
+
+A. Backport Provider Packages (those are needed in 1.10.* Airflow series) are going to be released for
+   3 months after the release. We will stop releasing new updates to the backport providers afterwards.
+   You will ba able to continue using the provider packages that you already use and unless you need to
+   get some new release of the provider that is only released in master, there is no need to upgrade
```

Comment:
What would be a case in which this happens (ie. a provider is only released in master)? If the BaseOperator changes or something?

Reply:
Clarified. Not really. We expect some future changes to providers. A good example might be Google Python client 2.0.*. Google released 2.0 python clients recently for most of their services. Those are not backwards compatible. Most of the google providers for 2.0 will eventually migrate to those libraries and it's rather unlikely this will happen before the 3 months pass after Airflow 2.0, so those providers based on new clients will only be available for Airflow 2.0 (when the providers migrate to it) because we will not release backport providers then.